src/main/java/net/fabricmc/loom/configuration/ide/RunConfigSettings.java

```diff
@@ -0,0 +1,279 @@
+
+		mode(baseName);
+		source("main");
+		runDir("run");
+	}
+
+	public Project getProject() {
+		return project;
+	}
+
+	public LoomGradleExtension getExtension() {
+		return extension;
+	}
+
+	@Override
+	public String getName() {
+		return baseName;
+	}
+
+	public List<String> getVmArgs() {
+		return vmArgs;
+	}
+
+	public List<String> getProgramArgs() {
+		return programArgs;
+	}
+
+	public String getMode() {
+		return mode;
+	}
+
+	public void setMode(String mode) {
+		this.mode = mode;
+	}
+
+	public String getConfigName() {
+		return name;
+	}
+
+	public void setConfigName(String name) {
+		this.name = name;
+	}
+
+	public String getDefaultMainClass() {
+		return defaultMainClass;
+	}
+
+	public void setDefaultMainClass(String defaultMainClass) {
+		this.defaultMainClass = defaultMainClass;
+	}
+
+	public String getRunDir() {
+		return runDir;
+	}
+
+	public void setRunDir(String runDir) {
+		this.runDir = runDir;
+	}
+
+	public SourceSet getSource(Project proj) {
+		return source.apply(proj);
+	}
+
+	public void setSource(SourceSet source) {
+		this.source =
+				proj -> source;
+	}
+
+	public void setSource(Function<Project, SourceSet> sourceFn) {
+		this.source = sourceFn;
+	}
+
+	public void mode(String mode) {
+		setMode(mode);
+	}
+
+	public void name(String name) {
+		setConfigName(name);
+	}
+
+	public void defaultMainClass(String cls) {
+		setDefaultMainClass(cls);
+	}
+
+	public void runDir(String dir) {
+		setRunDir(dir);
+	}
+
+	public void vmArg(String arg) {
+		vmArgs.add(arg);
+	}
+
+	public void vmArgs(String... args) {
+		vmArgs.addAll(Arrays.asList(args));
+	}
+
+	public void vmArgs(Collection<String> args) {
+		vmArgs.addAll(args);
+	}
+
+	public void property(String name, String value) {
+		vmArg("-D" + name + "=" + value);
+	}
+
+	public void property(String name) {
+		vmArg("-D" + name);
+	}
+
+	public void properties(Map<String, String> props) {
+		props.forEach(this::property);
+	}
+
+	public void programArg(String arg) {
+		programArgs.add(arg);
+	}
+
+	public void programArgs(String... args) {
+		programArgs.addAll(Arrays.asList(args));
+	}
+
+	public void programArgs(Collection<String> args) {
+		programArgs.addAll(args);
+	}
+
+	public void source(SourceSet source) {
+		setSource(source);
+	}
+
+	public void source(String source) {
+		setSource(proj -> {
+			JavaPluginConvention conv = proj.getConvention().getPlugin(JavaPluginConvention.class);
+			return conv.getSourceSets().getByName(source);
+		});
+	}
+
+	/**
+	 * Add the {@code -XstartOnFirstThread} JVM argument when on OSX.
+	 */
+	public void startFirstThread() {
+		if (OperatingSystem.getOS().equalsIgnoreCase("osx")) {
+			vmArg("-XstartOnFirstThread");
+		}
+	}
+
+	/**
+	 * Removes the {@code nogui} argument for the server configuration. By default {@code nogui} is specified, this is
+	 * a convenient way to remove it if wanted.
+	 */
+	public void serverWithGui() {
+		while (programArgs.remove("nogui")) {
+			// N/A, the condition does the work
+		}
```

Comment:
```suggestion
		programArgs.removeIf("nogui"::equals);
```
Removes all occurrences in one line